### DISCUSSION TITLE
Is  it possible to use entityviews as spring data projections?

### DISCUSSION THREAD

**Discussion by bukajsytlos** [Blazebit/blaze-persistence#1790] [General]
From what I've quickly observed, entity views are "considered" as separate entities to original entities, and have to have own repositories and so on.
I am looking for sql wise effective projections that can be easily requested against existing repositories. Is that somehow possible?
Thanks

**Comment by beikov** [comment]
>From what I've quickly observed, entity views are "considered" as separate entities to original entities, and have to have own repositories and so on.

If you [`@EnableBlazeRepositories`](https://staging-persistence.blazebit.com/documentation/1.6/entity-view/manual/en_US/index.html#spring-data-setup) for you current repositories, then you can use entity views within your "regular" repositories. Just change the return type of some method or use the dynamic projection style `<T> List<T> findAll(Class<T> clazz)`.
